File: market-application/src/main/java/com/example/market/application/port/in/PlaceBidUseCase.java
package com.example.market.application.port.in;

import com.example.market.application.command.PlaceBidCommand;
import com.example.market.domain.trading.BidId;
import com.example.market.domain.trading.TradeId;

import java.util.Optional;

public interface PlaceBidUseCase {

    PlaceBidResult place(PlaceBidCommand command);

    record PlaceBidResult(BidId bidId, Optional<TradeId> matchedTradeId) {}
}

File: market-application/src/main/kotlin/com/example/market/application/port/in/PlaceBidUseCase.kt
package com.example.market.application.port.`in`

import com.example.market.application.command.PlaceBidCommand
import com.example.market.domain.trading.BidId
import com.example.market.domain.trading.TradeId
import java.util.Optional

interface PlaceBidUseCase {

    fun place(command: PlaceBidCommand): PlaceBidResult

    @JvmRecord
    data class PlaceBidResult(val bidId: BidId, val matchedTradeId: Optional<TradeId>)
}
